controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -273,18 +172,26 @@ func (w *Workload) RemoveNodeFromKubeadmConfigMap(ctx context.Context, name stri
 			Log.Error(err, "unable to get kubeadmConfigMap")
 			return false, nil
 		}
-		config := &kubeadmConfig{ConfigMap: kubeadmConfigMap}
+		config := NewKubeadmConfig(kubeadmConfigMap)
 		if err := config.RemoveAPIEndpoint(name); err != nil {
 			return false, err
 		}
-		if err := w.Client.Update(ctx, config.ConfigMap); err != nil {
+		if err := w.Client.Update(ctx, config.GetConfigMap()); err != nil {
```

Comment:
Wondering if we should use the patch helper consistently instead of update...
@vincepri opinions?

Reply:
IMO, Patch makes more sense if there are multiple updates being made to the same object, like in the case of Reconcile loops. This one is a little different, but one of the places where this gets called makes it inefficient since it is making update calls in a loop, which could easily be morphed into a single Patch/Update call.

But the point of this method itself is so that it can be reused in multiple places, and taking out the update would mean that we would need to add the Patch functionality in multiple places where this method is being invoked from.

Reply:
I'm fine in keeping the current approach, but in the long term I would like to use consistently the patch helper vs deciding case by case